In [ ]:
import pandas as pd
df = pd.read_csv("C:/Users/rodri/Desktop/Projects/healthcare-performance-analytics/data/raw/healthcare-data.csv", sep=";")
print(df)

          periodo                   ars  \
0      2026-09-19             ARS Norte   
1      2026-09-19            ARS Centro   
2      2026-09-19            ARS Centro   
3      2026-09-19            ARS Centro   
4      2026-09-19          ARS Alentejo   
...           ...                   ...   
84672  2016-11-01           ARS Algarve   
84673  2016-11-01             ARS Norte   
84674  2016-11-01  Portugal Continental   
84675  2016-11-01          ARS Alentejo   
84676  2016-11-01  Portugal Continental   

                                               indicador       valor  \
0      Taxa diária de atendimentos urgentes com prior...    30.85170   
1      Tempo médio de espera entre a triagem e a prim...    69.55160   
2      Taxa diária de atendimentos urgentes com prior...    29.12730   
3               Número estimado de episódios de urgência  2671.00000   
4      Taxa diária de atendimentos urgentes com prior...    45.71010   
...                                                

In [7]:
#Basic Overview 
print("\n SHAPE")
print(df.shape)

print("\n COLUMNS")
print(df.columns)

print("\n DATA TYPES")
print(df.dtypes)

print("\n NULL VALUES")
print(df.isnull().sum())

print("\n DUPLICATED VALUES")
print(df.duplicated().sum())


 SHAPE
(84677, 6)

 COLUMNS
Index(['periodo', 'ars', 'indicador', 'valor', 'unidade', 'id'], dtype='str')

 DATA TYPES
periodo      datetime64[us]
ars                     str
indicador               str
valor               float64
unidade                 str
id                      str
dtype: object

 NULL VALUES
periodo      0
ars          0
indicador    0
valor        0
unidade      0
id           0
dtype: int64

 DUPLICATED VALUES
0


Aqui percebemos que existem 84677 registos e 6 colunas. As colunas são periodo ars, indicador, valor, unidade, id. Não existem valores nulos e também não existem valores duplicados, o que facilita a nossa limpeza no próximo script.

In [6]:
print("\n UNIQUE ARS ")
print(df["ars"].unique())

print("\n NUMBER OF ARS")
print(df["ars"].nunique())

print("\n UNIQUE INDICATORS")
print(df["indicador"].unique())

print("\n NUMBER OF INDICATORS")
print(df["indicador"].nunique())

print("\n UNIQUE UNITS")
print(df["unidade"].unique())



 UNIQUE ARS 
<StringArray>
[                'ARS Norte',                'ARS Centro',
              'ARS Alentejo',      'Portugal Continental',
 'ARS Lisboa e Vale do Tejo',               'ARS Algarve']
Length: 6, dtype: str

 NUMBER OF ARS
6

 UNIQUE INDICATORS
<StringArray>
[                                 'Taxa diária de atendimentos urgentes com prioridade verde ou azul',
 'Tempo médio de espera entre a triagem e a primeira observação médica (rede de urgência hospitalar)',
                                                           'Número estimado de episódios de urgência',
                                              'Taxa diária de atendimentos urgentes com internamento']
Length: 4, dtype: str

 NUMBER OF INDICATORS
4

 UNIQUE UNITS
<StringArray>
['epis. urg./100.000 resid.', 'minuto', 'episódio urgência', '%']
Length: 4, dtype: str


Com este overview percebemos que existem 6 Administrações Regionais de Saúde , 4 tipos de indicadores diferentes e 4 unidades distintas também.

In [5]:
# Date Range
df["periodo"] = pd.to_datetime(df["periodo"])
print("\n DATE RANGE")
print("start:", df["periodo"].min())
print("end:", df["periodo"].max())



 DATE RANGE
start: 2016-11-01 00:00:00
end: 2026-09-19 00:00:00


Aqui percebemos que o primeiro registo foi dia 01-11-2016 e o último foi dia 19-09-2026 , ou seja os dados compreendem um intervalo de 10 anos

In [8]:
# Records Per Region
print("\n RECORDS PER ARS")
print(df["ars"].value_counts())

#Records per Indicator
print("\n RECORDS PER INDICATOR")
print(df["indicador"].value_counts())


 RECORDS PER ARS
ars
ARS Norte                    14368
ARS Centro                   14368
ARS Alentejo                 14368
Portugal Continental         14368
ARS Lisboa e Vale do Tejo    14368
ARS Algarve                  12837
Name: count, dtype: int64

 RECORDS PER INDICATOR
indicador
Número estimado de episódios de urgência                                                              21552
Taxa diária de atendimentos urgentes com prioridade verde ou azul                                     21144
Taxa diária de atendimentos urgentes com internamento                                                 21143
Tempo médio de espera entre a triagem e a primeira observação médica (rede de urgência hospitalar)    20838
Name: count, dtype: int64


Temos aqui infromações dos group by, e conseguimos perceber que o ARS Algarve tem menos registos. O que pode indicar que existem falhas ao longo dos anos em determinados indicadores.